app/src/main/java/com/babylon/wallet/android/presentation/onboarding/restore/mnemonics/RestoreMnemonicsViewModel.kt

```diff
@@ -67,11 +67,15 @@ class RestoreMnemonicsViewModel @Inject constructor(
             } ?: run {
                 getProfileUseCase.flow.firstOrNull()
             }
-            val factorSources = profile.recoverableFactorSources()
+            val mainBabylonFactorSourceId = profile?.mainBabylonFactorSource?.value?.id?.asGeneral()
+            // we want main factor source to go first
+            val factorSources = profile.recoverableFactorSources().sortedByDescending {
+                it.factorSource.value.id.asGeneral() == mainBabylonFactorSourceId
```

Comment:
I am not sure I follow how this function works. Do we expect to sort somehow by the id bytes and the kind of the factor source? Isn't this a bit random? We only wanted to put main first.

Reply:
Now that I read the doc for this method, I suppose that it sorts based on this selector. Anything different from it will keep its order. So I think it looks good.